nutils/sample.py

```diff
@@ -0,0 +1,462 @@
+:func:`nutils.topology.Topology.sample`. Any function evaluation starts from
+this sampling step, which drops element information and other topological
+properties such as boundaries and groups, but retains point positions and
+(optionally) integration weights. Evaluation is performed by subsequent calls
+to :func:`Sample.integrate`, :func:`Sample.integral` or :func:`Sample.eval`.
+
+Besides the location of points, ``Sample`` also keeps track of point
+connectivity through its :attr:`Sample.tri` and :attr:`Sample.hull`
+properties, representing a (n-dimensional) triangulation of the interior and
+boundary, respectively. Availability of these properties depends on the
+selected sample points, and is typically used in combination with the "bezier"
+set.
+
+In addition to `Sample`, the sample module defines the :class:`Integral` class
+which represents postponed integration. Integrals are internally represented as
+pairs of :class:`Sample` and :class:`nutils.function.Array` objects. Evaluation
+proceeds via either the :func:`Integral.eval` method, or the
+:func:`eval_integrals` function. The latter can also be used to evaluate
+multiple integrals simultaneously, which has the advantage that it can
+efficiently combine common substructures.
+'''
+
+from . import types, points, log, util, function, config, parallel, numeric, cache, matrix
+import numpy, numbers
+
+class Sample(types.Singleton):
+  '''Collection of points on a topology.
+
+  The :class:`Sample` class represents a collection of discrete points on a
+  topology and is typically formed via :func:`nutils.topology.Topology.sample`.
+  Any function evaluation starts from this sampling step, which drops element
+  information and other topological properties such as boundaries and groups,
+  but retains point positions and (optionally) integration weights. Evaluation
+  is performed by subsequent calls to :func:`integrate`, :func:`integral` or
+  :func:`eval`.
+
+  Besides the location of points, ``Sample`` also keeps track of point
+  connectivity through its :attr:`tri` and :attr:`hull` properties,
+  representing a (n-dimensional) triangulation of the interior and boundary,
+  respectively. Availability of these properties depends on the selected sample
+  points, and is typically used in combination with the "bezier" set.
+  '''
+
+  @types.apply_annotations
+  def __init__(self, transforms:tuple, points:types.tuple[points.strictpoints], index:types.tuple[types.frozenarray[types.strictint]]):
+    '''Constructor.
+
+    Args
+    ----
+    transforms : :class:`tuple` or transformation chains
+        List of transformation chains leading to local coordinate systems that
+        contain points.
+    points : :class:`tuple` of point sets
+        List of point sets matching ``transforms``.
+    index : :class:`tuple` of integer arrays
+        List of indices matching ``transforms``, defining the order on which
+        points show up in the evaluation.
+    '''
+
```

Comment:
Like in `points.Point` this docstring is ignored by sphinx. Can we merge this with the class docstring?